servicetalk-transport-netty-internal/src/main/java/io/servicetalk/transport/netty/internal/DefaultNettyConnection.java

```diff
@@ -941,6 +919,7 @@ public void channelActive(ChannelHandlerContext ctx) {
 
         @Override
         public void channelInactive(ChannelHandlerContext ctx) {
+            connection.notifyOnClosing();
```

Comment:
if a channel was never active this method may not be called (e.g. just handler removed) do we need to also invoke this method in `handlerRemove`?

Reply:
I think it's enough to notify only from `channelInactive`. If channel was never active, and we close locally, the `channel.closeFuture()` or `closeAsync/closeAsyncGracefully` will execute first. This is here specifically for the case when channel was active but unexpectedly closed by the server and for some reason we haven't seen `ChannelInputShutdownReadComplete` before `channelInactive`. IIRC, it can happen with Java NIO networking.